### ISSUE TITLE
Implement C++ GDAL driver

### ISSUE THREAD

**Issue by mpadge** [r-gris/gladr#1]
okay, let's use this as an initial repo to play around in. Basic reading code [here](http://gdal.org/ogr_apitut.html) and driver implementations [here](http://gdal.org/ogr_drivertut.html). As they say at the end of the latter,
> There! We have completed a simple read-only feature file format driver.

The rest is just sugar on top. I might not actually find time to do this for a little while, but will place it high on my priority to-do list ...

**Comment by mdsumner** [comment]
@mpadge I only really need a Linux-workable  barest example in R package form and I'll be able to crawl along slowly from there. That's perhaps a simpler first task? Even if the include paths are hard-coded and so on - those infrastructure basics are what I get stuck on.  

 I am actually pretty fine with the GDAL API itself, I've used it in C++ raw and in C# and in python,  but boot-strapping into that with a package leaves me totally lost in how Makevars and configure operate  (I understand vaguely that they are cross-platform ways to specify where the headers and libs are ...)

I've done this for myself with NetCDF here, and that will keep me busy for weeks relating the API to how I understand it from R already, then later I can pursue "proper standards" for the set up: https://github.com/hypertidy/ncapi  I was stuck on that for ages but now I have it all building I can prod and poke my way forward. That has given me better focus on what I need to ask assistance with. :)

**Comment by mpadge** [comment]
thanks mike - i had a vague notion in my head of finding some time next week to give this a crack. I've got a rough draft somewhere, but can't recall whether I wrapped it in an R package or not. I'll get on to it asap.

**Comment by mdsumner** [comment]
I've finally figured it out. I'm at a very primitive stage, no error handling and hard coded files and so on - but this is enough to learn from already!  The crux was autconf, which I started reading about but then just copied the sf/configure.ac, ran autoconf and then it worked. 

I'll hack away until I've got something half-way useable and then I'll establish a proper home and get you to review/fix the set up. :)
